File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        
        vector<vector<string>> v;
        map<string,vector<string>>mp;
        int n=strs.size();
        for(int i=0;i<n;i++)
        {
            string s=strs[i];
            sort(s.begin(),s.end());
            mp[s].push_back(strs[i]);
            
        }
        for(auto itr=mp.begin();itr!=mp.end();itr++)
        {
            v.push_back(itr->second);
            
        }
        return v;
    }
};

File: 0049-group-anagrams/0049-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {

        Map<String, List<String>> mp = new HashMap<>();

        for (int i = 0; i < strs.length; i++) {

            char[] temp = strs[i].toCharArray(); 
            Arrays.sort(temp);
            String key = new String(temp);

            mp.putIfAbsent(key, new ArrayList<>());
            mp.get(key).add(strs[i]);
        }

        List<List<String>> ans = new ArrayList<>();

        for (List<String> group : mp.values()) {
            ans.add(group);
        }

        return ans;
    }
}
